File: src/BuildingBlocks/nodejs/framework/bus/abstratctions/saga.ts
import { Command } from '../../infrastructure';
import { ICommandBus } from './iBus';


export  class Saga {
    public CommandBus: ICommandBus;
    constructor(commandBus: ICommandBus) {
        //check for null
        if(!commandBus){
            throw 'Command bus is null';
        }
        this.CommandBus = commandBus;
    }

    public Handle(message:Command): void {

    }
}

File: src/BuildingBlocks/dotnet/Framework/Bus/Abstractions/Saga.cs
using Microsoft.EntityFrameworkCore;
using Monad.EDukaan.Framework.Common.Entities;
using Monad.EDukaan.Framework.Persistence.Domain.Interfaces;
using Monad.EDukaan.Service.Identity.Infrastructure;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Monad.EDukaan.Framework.Bus.Abstractitons
{
    public abstract class Saga
    {
        public ICommandBus CommandBus { get; set; }
        public Saga(ICommandBus bus)
        {

        }
    }
}